### PR TITLE
Use upstream image builds

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/assign @smarterclayton 
/cc @petr-muller @bbguimaraes @droslean 

/hold

### CODE REVIEW

File: hack/bump-prow-images.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+latest_tag="$( gcloud container images list-tags gcr.io/k8s-prow/plank --format='value(tags)' --limit 1 | grep -Po "v[0-9]+\-[a-z0-9]+" )"
```
Comment (smarterclayton) [inline hack/bump-prow-images.sh]: tiny doc in here explaining what this is doing

File: cluster/ci/config/prow/openshift/plank.yaml [hunk 2]
```diff
@@ -1,34 +1,38 @@
 kind: List
 apiVersion: v1
 items:
-- apiVersion: apps.openshift.io/v1
-  kind: DeploymentConfig
+- apiVersion: extensions/v1beta1
```
Comment (smarterclayton) [inline cluster/ci/config/prow/openshift/plank.yaml]: you should be able to do apps/v1

File: cluster/ci/config/prow/openshift/needs_rebase.yaml [hunk 3]
```diff
@@ -4,36 +4,40 @@ items:
...
+          component: needs-rebase
       spec:
         containers:
-        - name: "needs-rebase"
-          image: " "
+        - name: needs-rebase
+          image: gcr.io/k8s-prow/needs-rebase:v20181129-c3f22b5
           args:
-          - --github-token-file=/etc/github/oauth
+          - --github-token-path=/etc/github/oauth
```
Comment (smarterclayton) [inline cluster/ci/config/prow/openshift/needs_rebase.yaml]: ugh

File: hack/bump-prow-images.sh (hunk 1 above)
Comment (stevekuznetsov) [reply hack/bump-prow-images.sh]: naaaah

Comment (stevekuznetsov) [reply hack/bump-prow-images.sh]: what's a doc?

File: cluster/ci/config/prow/openshift/needs_rebase.yaml (hunk 3 above)
Comment (stevekuznetsov) [reply cluster/ci/config/prow/openshift/needs_rebase.yaml]: /shrug

File: cluster/ci/config/prow/openshift/plank.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply cluster/ci/config/prow/openshift/plank.yaml]: All of the deployments I see on the cluster are `extensions/v1beta1`

File: ci-operator/jobs/infra-periodics.yaml [hunk 4]
```diff
@@ -31,7 +31,7 @@ periodics:
       - --confirm
       command:
       - /bin/commenter
-      image: docker-registry.default.svc:5000/ci/commenter:latest
+      image: gcr.io/k8s-testimages/commenter:v20181130-d7867ec
```
Comment (petr-muller) [inline ci-operator/jobs/infra-periodics.yaml]: Hm, will we need to change the version tags everywhere when we want to bump? How often do we bump?

**Comment by stevekuznetsov** [comment]
Upstream stopped allowing image builds using Dockerfiles and all the normal mechanisms, so we'd have to build up a parallel build pipeline to use `bazel`...

**Comment by petr-muller** [comment]
Ah, now I remember you mentioning that. Avoiding setting up a `bazel` pipeline sounds like a good thing.

**Comment by stevekuznetsov** [comment]
This went way more smoothly than I expected.

**Comment by stevekuznetsov** [comment]
Famous last words

**Pull request merged**